src/interlocking_component/route_controller.py

```diff
@@ -95,15 +74,58 @@ def insert_train_priority_changed(self, train: Train):
         raise NotImplementedError()
 
 
-class RouteController(IRouteController):
+class RouteController:
     """This class coordinates the route of a train.
     It calls the router to find a route for a train.
     It makes sure, that the Interlocking sets fahrstrassen along those routes.
     """
 
-    interlocking: interlockinginterface = None
+    interlocking: Interlocking = None
```

Comment:
What's your plan for assigning this variable? Do you want to assign it in a constructor or just "set" it right away? 🙃

Reply:
Does that mean you'll initialize or pass & set it in the constructor? 🙃

Reply:
Ah nevermind 😂  I was just blind 👍 Perfect Thank you!